### DISCUSSION TITLE
How to use this with a react native app?

### DISCUSSION THREAD

**Discussion by emilnili** [aws-geospatial/amazon-location-utilities-auth-helper-js#25] [Q&A] [answered]
We have set up a place index and have been testing this from our backend.
It worked fine when running the code locally. But when we want to implement this into our Lambda the problems arise.

It seems the @aws-sdk/client-location will have dependencies all over the place even react-native deps, causing the lambda to explode in size. And this utilities have dps on client-location.

So that did no work.

Then we tried doing all this in the client side instead, using apiKey. So we added these to the client:
@aws-sdk/client-location
@aws/amazon-location-utilities-auth-helper

Seems fine, BUT when we try to run it. It will cause errors saying some of the deps from these are causing this:
`The package at "node_modules/@smithy/util-stream/dist-cjs/getAwsChunkedEncodingStream.js" attempted to import the Node standard library module "stream".
It failed because the native React runtime does not include the Node standard library.`

If we add these also we will run into more and more problems.

So how are we supposed to use this for a react native app? Or in a nodejs aws lambda?

**Comment by cgalvan** [comment, marked as answer]
Hello @emilnili 

I merged a change recently that re-worked the dependencies to reduce the dependencies size from 280MB -> 19MB:

https://github.com/aws-geospatial/amazon-location-utilities-auth-helper-js/pull/35

I pushed a new release here: https://github.com/aws-geospatial/amazon-location-utilities-auth-helper-js/releases/tag/v1.0.6

This also eliminated the react-native dependencies. Can you see if you can use this now from your node lambda?

If you wanted to use it in a react-native app, you should be able to bundle it now that the size is greatly reduced.

Please ping me if you run into any issues.

**Reply by cgalvan** [reply]
Please re-open this discussion if my suggestion to use the latest version doesn't work for you, and I'll be happy to investigate!
